ggshield/__main__.py

```diff
@@ -138,9 +139,14 @@ def _set_color(ctx: click.Context):
         ctx.color = True
 
 
-def _display_deprecation_message(cfg: Config) -> None:
-    for message in cfg.user_config.deprecation_messages:
+def _display_deprecation_message(ctx_obj: ContextObj) -> None:
+    for message in ctx_obj.config.user_config.deprecation_messages:
         ui.display_warning(message)
+    if sys.version_info < (3, 9) and ctx_obj.output_format is OutputFormat.TEXT:
```

Comment:
Actually, I added this `ctx_obj.output_format is OutputFormat.TEXT` clause because otherwise it breaks the json outputs. The warning message ends up concatenated to the json which can't be parsed anymore

Reply:
Is this happening inside a test? If this is the case then the test must be changed to use `cli_fs_runner.mix_stderr = False`. You can find an example here: https://github.com/GitGuardian/ggshield/blob/main/tests/unit/cmd/scan/test_path.py#L109.

Reply:
Yes, it was. Thanks for your input, I'm going to try it

Reply:
see ae62e544a71366f408099b8dc5f93429a898f9dd